packages/block-editor/src/components/link-control/index.js

```diff
@@ -94,12 +78,8 @@ function LinkControl( {
 
 		// Populate input searcher whether
 		// the current link has a title.
-		if ( currentLink && currentLink.title ) {
-			setInputValue( currentLink.title );
-		}
-
-		if ( isFunction( onChangeMode ) ) {
-			onChangeMode( mode );
+		if ( value && value.title && mode === 'edit' ) {
+			setInputValue( value.title );
```

Comment:
I'm not convinced we should be using title here and this actually reveals a fundemental flaw in our interface.

If you search for `Hello world` and it returns a search result for a `Page` with the title "Hello world". You then select this. You then want to edit that link so you click `Change`. What do you expect to see?

1. The original text "Hello world" in the search box (presumably with the original search result shown below as well - this doesn't currently happen!).
2. A blank search box - you wanted to change it after all. 
3. The URL of the link itself.

It's not clear to me.

Personally I'm thinking that we ought to have a separate `Edit` button which affords the ability to edit the raw hyperlink itself. 

This would also solve the issue which will occur for manually entering URLs when we start displaying the `<title>` of the remote website in the search results. Currently, this `title` text would be the `value.title`. Therefore when clicking change the user would be presented with the text of the remote website's `title` text and not the initial URL they entered (which is what they would expect - after all the `title text` won't return any search results).

This might not need to be fixed in this PR but it's worth noting.

Reply:
Had the same reflexion, setting the URL as a starting point seemed more logic somehow.